readthedocs/projects/tasks.py

```diff
@@ -671,30 +668,27 @@ def run_setup(self, record=True):
 
         # Environment used for code checkout & initial configuration reading
         with environment:
+            before_vcs.send(sender=self.version, environment=environment)
+            if self.project.skip:
+                raise ProjectBuildsSkippedError
             try:
-                before_vcs.send(sender=self.version, environment=environment)
```

Comment:
The outer try..except was removed and the content was dedented, not other changes were made, the diff looks weird :D